events/syscall/read_ssh_information.go

```diff
@@ -0,0 +1,51 @@
+//go:build linux
+// +build linux
+
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2023 The Falco Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package syscall
+
+import (
+	"path/filepath"
+
+	"github.com/falcosecurity/event-generator/events"
+
+	"os"
+)
+
+var _ = events.Register(
+	ReadSshInformation,
+	events.WithDisabled(), // the rule is not enabled by default, so disable the action too
+)
+
+func ReadSshInformation(h events.Helper) error {
+	// Also creates .ssh directory inside tempDirectory
+	tempDirectoryName, err := createSshDirectoryUnderHome()
+	if err != nil {
+		return err
+	}
+	sshDir := filepath.Join(tempDirectoryName, ".ssh")
```

Comment:
@FedeDP Yes we can do that! But before the function return we should also clean the temp folder created by createSshDirectoryUnderHome .
 `defer os.RemoveAll(tempDirectoryName)`
So if we return sshDir instead of tempDir . How can we clean tempDir ?

Reply:
So if we return sshdir then one valid approach is trimming the last 5 characters in sshDir gives tempdir path. Is that ok?

Reply:
I got it, your are right! Nope, i'd leave another comment on the helper function!